src/controllers/apiController.ts

```diff
@@ -1,21 +1,24 @@
-import { Request, Response, NextFunction } from 'express';
+import { NextFunction, Request, Response } from 'express';
 import Collection from '../models/2014/collection/index.js';
 
-export const index = async (req: Request, res: Response, next: NextFunction) => {
+export default async (req: Request, res: Response, next: NextFunction) => {
   try {
-    const data = await Collection.find({})
-      .select({ index: 1, _id: 0 })
-      .sort({ index: 'asc' })
-      .exec();
+    const collections = await Collection.find({});
 
-    const apiIndex: Record<string, string> = { '2014': '/api/2014' };
-    data.forEach((item) => {
-      if (item.index === 'levels') return;
+    const colName = req.path.split('/')[1];
+    const colRequested = collections.find(col => col.index === colName);
 
-      apiIndex[item.index] = `/api/${item.index}`;
-    });
-
-    return res.status(200).json(apiIndex);
+    if (colRequested === undefined && colName !== '') {
+      res.sendStatus(404);
+      return;
+    }
+    
+    const queryString = req.originalUrl.split('?')?.[1];
+    const redirectUrl = '/api/2014' + req.path;
```

Comment:
Wait, shouldn't this be everything after the split like in line 8? I assume this would just give you `/api/2014/api/spells/acid-arrow`.

Reply:
`req.path` is the bit of the URL that sits _after_ the route that was matched to get us to this controller (sans the query string). So because this controller sits under `/api`, with the pattern `"*"` for the handler itself, a request to `/api/spells/acid-arrow` will have `req.path === "/spells/acid-arrow".

So `req.path.split('/')[1] === 'spells'`, and `'/api/2014' + req.path === '/api/2014/spells/acid-arrow'`. The reason it's `[1]` is because of the leading `'/'` in the path, which will always be present (as express represents the "empty" path as `'/'`, so the "first" path segment will be at index `1`.

This is validated by the unit & integration tests :)